### DISCUSSION TITLE
How to propagate edge weights from original graph as node signal to line graph

### DISCUSSION THREAD

**Discussion by skmda37** [networkx/networkx#5880] [Q&A]
In the documentation of the `line_graph` method (https://networkx.org/documentation/stable/reference/generated/networkx.generators.line.line_graph.html)  it says 

> Graph, node, and edge data are not propagated to the new graph.

I specifically want to get the edge weights of a graph G as the node signal of line graph `line_graph(G)`. 

Does somebody have an idea on how to do that?

**Comment by rossbar** [comment]
Check out the last example in the `Examples` section of the docstring:

https://networkx.org/documentation/latest/reference/generated/networkx.generators.line.line_graph.html

It shows how you can manually copy over edge attributes from the original graph as node attributes in the line graph.
